pkg/scheduler/core/generic_scheduler.go

```diff
@@ -215,26 +211,31 @@ func (g *genericScheduler) assignReplicas(clusters []*clusterv1alpha1.Cluster, r
 	targetClusters := make([]workv1alpha2.TargetCluster, len(clusters))
 
 	if object.Replicas > 0 && replicaSchedulingStrategy != nil {
-		if replicaSchedulingStrategy.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDuplicated {
+		switch replicaSchedulingStrategy.ReplicaSchedulingType {
+		case policyv1alpha1.ReplicaSchedulingTypeDuplicated:
 			for i, cluster := range clusters {
 				targetClusters[i] = workv1alpha2.TargetCluster{Name: cluster.Name, Replicas: object.Replicas}
 			}
 			return targetClusters, nil
-		}
-		if replicaSchedulingStrategy.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDivided {
-			if replicaSchedulingStrategy.ReplicaDivisionPreference == policyv1alpha1.ReplicaDivisionPreferenceWeighted {
+		case policyv1alpha1.ReplicaSchedulingTypeDivided:
+			switch replicaSchedulingStrategy.ReplicaDivisionPreference {
+			case policyv1alpha1.ReplicaDivisionPreferenceWeighted:
+				// If ReplicaDivisionPreference is set to "Weighted" and WeightPreference is not set,
+				// scheduler will weight all clusters averagely.
 				if replicaSchedulingStrategy.WeightPreference == nil {
-					// if ReplicaDivisionPreference is set to "Weighted" and WeightPreference is not set, scheduler will weight all clusters the same.
 					replicaSchedulingStrategy.WeightPreference = getDefaultWeightPreference(clusters)
 				}
-				return g.divideReplicasByStaticWeight(clusters, replicaSchedulingStrategy.WeightPreference.StaticWeightList, object.Replicas)
-			}
-			if replicaSchedulingStrategy.ReplicaDivisionPreference == policyv1alpha1.ReplicaDivisionPreferenceAggregated {
-				return g.divideReplicasAggregatedWithResource(clusters, object)
+				if len(replicaSchedulingStrategy.WeightPreference.DynamicWeight) != 0 {
```

Comment:
Hi, if `DynamicWeight` policy is set, but the resource's `request` is not set. The maxAvailableReplicas will be `allowedPodNumber := resourceSummary.Allocatable.Pods().Value() - resourceSummary.Allocated.Pods().Value() - resourceSummary.Allocating.Pods().Value()`, right?

Reply:
That's it!